apps/angularlogos/src/app/shared/data.service.ts

```diff
@@ -15,49 +15,49 @@ export class DataService {
 
   constructor(private http: HttpClient) {}
 
-  getLogos() {
+  getLogos(): Observable<LogoEntry[]> {
     if (!this.logos$) {
       this.logos$ = forkJoin([this.http.get<any[]>(`${environment.apiBaseUrl}/contents/logos`), this.getMetadataCached()]).pipe(
-        map(([contents, metadataAll]) => contents.map((content) => this.contentToLogoEntry(content, metadataAll))),
+        map(([contents, metadataAll]) => contents.map((content) => this.mapGitHubContentToLogoEntry(content, metadataAll))),
         shareReplay()
       );
     }
 
     return this.logos$;
   }
 
-  getLogosWithoutMetadata() {
+  getLogosWithoutMetadata(): Observable<LogoEntry[]> {
     return this.getLogos().pipe(
       withLatestFrom(this.getMetadataCached()),
-      map(([logos, metadata]) => logos.filter((logo) => !metadata[logo.filename]))
+      map(([logos, metadata]) => logos.filter((logo) => !metadata[logo.fileName]))
     );
   }
 
-  getLogosFiltered(searchTerm: string) {
+  getLogosFiltered(searchTerm: string): Observable<LogoEntry[]> {
     const lowerCaseTerm = searchTerm.toLowerCase();
     return this.getLogos().pipe(
       map((logos) => (searchTerm ? logos.filter((logo) => logo.name.toLowerCase().includes(lowerCaseTerm)) : logos))
     );
   }
 
-  private contentToLogoEntry(content: any, metadataAll: LogoMetadataAll): LogoEntry {
-    const filename = content.name;
-    const metadata = metadataAll[filename] || {};
+  private mapGitHubContentToLogoEntry(content: any, metadataAll: LogoMetadataEntity): LogoEntry {
+    const fileName = content.name;
```

Comment:
I'd not use camel case here since filename can be written as one word: https://en.wikipedia.org/wiki/Filename

Reply:
There are a lot of opinions on that naming conventions, conclusion it comes down to flavour 😄